### DISCUSSION TITLE
For Apple silicon Mac

### DISCUSSION THREAD

**Discussion by djhan** [oomek/attractplus#13] [Show and tell]
Hello, I'm DJ.HAN.
I was forked your attractplus for Apple silicon Mac. I changed makefile, create-pkg.sh, bundlelibs.py and launch.sh. Please check link;

https://github.com/djhan/attractplus-for-mac

I found code-signing is mandatory on current MacOS. You must apply signing certificate to own binary and other libraries(dylib). 
Of course, this method can used on Intel Mac. But I don't know how classify CPU architecture of Mac on makefile source. 
Please check my source. Thanks, best regards.

**Comment by zpaolo11x** [comment]
This is great news @djhan for all macOS users (like me). I'll look into the code, but I have a couple of questions if you can answer:

- Did you use libraries from homebrew, and package them within the application, so the app bundle can run even if homebrew is not installed?
- I thought code signing was not mandatory as long as one can open the app with the right mouse button and force it to run even if it's from unknown developer. Is it still the case or am I missing something?

**Reply by djhan** [reply]
1) Yes, if you build with your codesign identity, you can run Attract-Plus without homebrew libraries.
2) Unfortunately, current MacOS needs code signing. If you build package without code signing, it will not work other's Mac. Not work right-mouse button trick. But if you found a way, please tell me.

**Reply by zpaolo11x** [reply]
>     2. Unfortunately, current MacOS needs code signing. If you build package without code signing, it will not work other's Mac. Not work right-mouse button trick. But if you found a way, please tell me.

Oh, I'm on Big Sur (late 2013 MBP) and the option is still there. What about this guide? https://www.howtoisolve.com/open-apps-from-an-unauthorized-developer-on-mac-os-x/   It seems on Ventura too you can try to run the application, and when it fails go into settings, privacy and security and enable the code from unknown developer.

**Reply by djhan** [reply]
The guide works only your own binary app. But attractplus package includes dylibs from homebrew, not your own binaries. If you packaging attractplus without code-signing, it will terminate instantly when you tried run.

**Reply by zpaolo11x** [reply]
Got it, when I made my own package for attractplus I had to manually open and "accept" all dylibs so I see the point here. Thank you!
